control-admin/src/main/java/fi/nls/oskari/control/admin/AbstractLayerAdminHandler.java

```diff
@@ -0,0 +1,140 @@
+package fi.nls.oskari.control.admin;
+
+import fi.nls.oskari.control.ActionDeniedException;
+import fi.nls.oskari.control.ActionException;
+import fi.nls.oskari.control.RestActionHandler;
+import fi.nls.oskari.domain.Role;
+import fi.nls.oskari.domain.User;
+import fi.nls.oskari.domain.map.OskariLayer;
+import fi.nls.oskari.log.LogFactory;
+import fi.nls.oskari.log.Logger;
+import fi.nls.oskari.service.OskariComponentManager;
+import fi.nls.oskari.service.ServiceException;
+import fi.nls.oskari.service.UserService;
+import fi.nls.oskari.util.JSONHelper;
+import fi.nls.oskari.util.PropertyUtil;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.oskari.permissions.PermissionService;
+import org.oskari.permissions.model.PermissionType;
+import org.oskari.permissions.model.Resource;
+import org.oskari.permissions.model.ResourceType;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+public abstract class AbstractLayerAdminHandler extends RestActionHandler {
+
+    private Logger log = LogFactory.getLogger(AbstractLayerAdminHandler.class);
+    private PermissionService permissionsService;
+    private UserService userService;
+
+    private String[] capabilitiesRoles;
+    private Set<String> availablePermissionTypes;
+
+
+
+    public void init() {
+        capabilitiesRoles = PropertyUtil.getCommaSeparatedList("actionhandler.GetWSCapabilitiesHandler.roles");
+        permissionsService = OskariComponentManager.getComponentOfType(PermissionService.class);
+        availablePermissionTypes = getAvailablePermissions();
+
+    }
+
+    protected PermissionService getPermissionsService() {
+        return permissionsService;
+    }
+
+    protected UserService getUserService() throws ServiceException {
+        if (userService == null) {
+            userService = UserService.getInstance();
```

Comment:
Why is this lazily initialized but PermissionService isn't?

Reply:
Now tries also in init. Reason for getUserService() is to handle ServiceException. Could also use UserService.getInstance() without getUserService() and initialization in AbstractLayerAdminHandler.